fdbclient/common.go

```diff
@@ -40,27 +43,41 @@ const (
 // DefaultCLITimeout is the default timeout for CLI commands.
 var DefaultCLITimeout = 10 * time.Second
 
-// getFDBDatabase opens an FDB database. The result will be cached for
-// subsequent calls, based on the cluster namespace and name.
-func getFDBDatabase(cluster *fdbv1beta2.FoundationDBCluster) (fdb.Database, error) {
-	clusterFile, err := os.CreateTemp("", "")
-	if err != nil {
-		return fdb.Database{}, err
-	}
+// createClusterFile will create or update the cluster file for the specified cluster.
+func createClusterFile(cluster *fdbv1beta2.FoundationDBCluster) (string, error) {
+	return ensureClusterFileIsPresent(os.TempDir(), string(cluster.UID), cluster.Status.ConnectionString)
+}
 
-	defer clusterFile.Close()
-	clusterFilePath := clusterFile.Name()
+// ensureClusterFileIsPresent will ensure that the cluster file with the specified connection string is present.
+func ensureClusterFileIsPresent(dir string, uid string, connectionString string) (string, error) {
+	clusterFileName := path.Join(dir, uid)
 
-	_, err = clusterFile.WriteString(cluster.Status.ConnectionString)
-	if err != nil {
-		return fdb.Database{}, err
+	// Try to read the file to check if the file already exists and if so, if the content matches
+	content, err := os.ReadFile(clusterFileName)
+
+	// If the file doesn't exist we have to create it
+	if errors.Is(err, fs.ErrNotExist) {
+		return clusterFileName, os.WriteFile(clusterFileName, []byte(connectionString), 0777)
+	}
+
+	// The content of the cluster file is already correct.
+	if string(content) == connectionString {
+		return clusterFileName, nil
 	}
-	err = clusterFile.Close()
+
+	// The content doesn't match, so we have to write the new content to the cluster file.
+	return clusterFileName, os.WriteFile(clusterFileName, []byte(connectionString), 0777)
+}
+
+// getFDBDatabase opens an FDB database. The result will be cached for
+// subsequent calls, based on the cluster namespace and name.
```

Comment:
```suggestion
// getFDBDatabase opens an FDB database.
```

We don't do the caching here but the FDB go bindings will do the caching for us implicitly.